multiuser/machine-auth/che-multiuser-machine-authentication/src/main/java/org/eclipse/che/multiuser/machine/authentication/server/MachineLoginFilter.java

```diff
@@ -121,20 +131,12 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
       } finally {
         EnvironmentContext.reset();
       }
-    } catch (UnsupportedJwtException
-        | MalformedJwtException
-        | SignatureException
-        | ExpiredJwtException ex) {
+    } catch (JwtException ex) {
       // signature check failed
       chain.doFilter(request, response);
```

Comment:
I think it is needed to `chain` **ONLY IF** it is not a machine token and throw an exception if it is a machine token and any exception occurs during token parsing/verification